packages/liveblocks-react-ui/src/components/Composer.tsx

```diff
@@ -706,3 +727,100 @@ export const Composer = forwardRef(
 ) as <M extends BaseMetadata = DM>(
   props: ComposerProps<M> & RefAttributes<HTMLFormElement>
 ) => JSX.Element;
+
+const ComposerRoomIdContext = createContext<string | null>(null);
+
+const ComposerSubmitCallback = createContext<
+  ((comment: ComposerSubmitComment) => void) | null
+>(null);
+
+function CreateThreadRoomIdProvider<M extends BaseMetadata>({
+  metadata,
+  children,
+}: ComposerCreateThreadProps<M> & { children: ReactNode }) {
+  const room = useRoom();
+  const createThread = useCreateThread();
+
+  const handleComposerSubmit = useCallback(
+    (comment: ComposerSubmitComment) => {
+      createThread({
+        body: comment.body,
+        metadata: metadata ?? {},
+        attachments: comment.attachments,
+      });
+    },
+    [createThread, metadata]
+  );
+
+  return (
+    <ComposerRoomIdContext.Provider value={room.id}>
+      <ComposerSubmitCallback.Provider value={handleComposerSubmit}>
+        {children}
+      </ComposerSubmitCallback.Provider>
+    </ComposerRoomIdContext.Provider>
+  );
+}
+
+function CreateOrEditCommentRoomIdProvider({
+  threadId,
+  commentId,
+  children,
+}: (ComposerCreateCommentProps | ComposerEditCommentProps) & {
+  children: ReactNode;
+}) {
+  const store = getUmbrellaStoreForClient(useClient());
+  const thread = store.getFullState().threadsDB.get(threadId);
+  if (thread === undefined) {
+    if (commentId === undefined) {
+      throw new Error(
+        `Cannot create a comment on thread ${threadId} because the thread does not exist in cache.`
+      );
+    } else {
+      throw new Error(
+        `Cannot edit comment ${commentId} on thread ${threadId} because the thread does not exist in cache.`
+      );
+    }
+  }
```

Comment:
@marcbouchenoire We throw an error if `Composer` is passed a thread id (and/or comment id) that doesn't exist in the cache. So,

```tsx
return <Composer threadId={"th_1"} commentId={"cm_1"} />
```

will throw an error if the thread id or comment id do not exist in cache.

A side-effect of this change is that a `Thread` component is also passed a thread data and if the associated thread doesn't exist in cache. So,

```tsx
return <Thread thread={thread} />
```

How do you feel about this change?